index.js

```diff
@@ -153,134 +52,81 @@ function getSha(imageName) {
+        PROJECTS,
+        project.ecrName,
+        project.headSha
       );
-      matchingProject.headSha = await getHeadSha(matchingProject.name);
-      matchingProject.headUrl = getLatestImageUrl(
-        matchingProject.ecrName,
-        matchingProject.headSha
+
+      const releaseContent = await getContents(
+        "notification-manifests",
+        project.manifestFile
       );
-      matchingProject.oldSha = getSha(image.newName);
-      matchingProject.oldUrl = image.newName;
-      image.newName = matchingProject.headUrl;
-      return matchingProject;
+
+      const originalFileContents = Base64.decode(releaseContent.content)
+      const re = new RegExp(`${project.ecrName}:\\S*`, "g");
+      project.oldUrl = originalFileContents.match(re)[0]
+      project.oldSha = getSha(project.oldUrl);
+      return project;
     })
   );
 }
 
+function UpdateContents(content, project) {
+  return content.replace(`${project.ecrName}:${project.oldSha}`, `${project.ecrName}:${shortSha(project.headSha)}`)
+}
+
 // Main ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 async function run() {
-  const releaseContent = await getContents(
-    GH_CDS,
-    "notification-manifests",
-    "env/production/kustomization.yaml"
-  );
-
   const prTemplate = await getContents(
-    GH_CDS,
     "notification-manifests",
     ".github/PULL_REQUEST_TEMPLATE.md"
   );
-
-  const releaseConfig = YAML.parse(Base64.decode(releaseContent.content));
   const issueContent = Base64.decode(prTemplate.content);
+  const manifestFiles = Array.from(new Set(PROJECTS.map(project => project.manifestFile)))
+
+  await hydrateWithSHAs();
+
+  var changesToManifestFiles = manifestFiles.map(async (manifestFile) => {
+    const projects = PROJECTS.filter(project => project.manifestFile == manifestFile)
+
+    const releaseContent = await getContents(
+      "notification-manifests",
+      manifestFile
+    );
+
+    var fileContents = Base64.decode(releaseContent.content)
+    projects.forEach((project) => {
+      fileContents = UpdateContents(fileContents, project)
+    })
+
+    const newReleaseContentBlob = Base64.encode(fileContents);
+
+    return { manifestFile, newReleaseContentBlob, releaseContent }
+  })
 
-  // Build up projects and update images with latest SHAs.
-  await hydrateWithSHAs(releaseConfig, PROJECTS);
+  changesToManifestFiles = await Promise.all(changesToManifestFiles)
 
   // Return if no new changes.
-  const newReleaseContentBlob = Base64.encode(YAML.stringify(releaseConfig));
-  if (newReleaseContentBlob.trim() === releaseContent.content.trim()) {
+  if (changesToManifestFiles.map(({ newReleaseContentBlob, releaseContent }) => {
+    newReleaseContentBlob.trim() === releaseContent.content.trim()
+  }).all) {
     return;
   }
 
   await closePRs();
-  await createPR(PROJECTS, issueContent, releaseContent, newReleaseContentBlob);
+  await createPR(PROJECTS, issueContent, changesToManifestFiles);
```

Comment:
That looks like an improvement to remove the concern of diffing the content from this `createPR` function. 👍